Add base and root bindings to pre test command

Ni-g-3l opened the pull request:
Hey ! 

I found that on pre_test_command, "base" and "root" bindings were missing. 

They are still available with "{this.base}" or "{this.root}", however I think it's important to keep consistency between available callback inside package.py

Feel free to comment or ask me question :)

maxnbk commented:
@Ni-g-3l , I've added tests to the PR that it would be nice if you could evaluate yourself, and if you could sign the CLA, we would be able to move forward merging this PR I believe.

Ni-g-3l commented:
Hi ! thanks ! I signed CLA but I don't know why it still doesn't work :/

JeanChristopheMorinPerso commented:
@Ni-g-3l it looks like commit https://github.com/AcademySoftwareFoundation/rez/commit/96313000df438735d1439885339fa6e0b499c6a9 was committed under a different email address. I think you can fix this using this:

```
git rebase --onto HEAD~2 --exec "git commit --amend --reset-author --no-edit" HEAD~2
```
(make sure that your git config is configured correctly. You can check using `git config user.email` and `git config user.name`).

JeanChristopheMorinPerso commented:
Also, it might be worth rebasing on main. I think the failure to build the docs is related to the branch being tool old. Lastly, I think we need to update the docs to reflect that there are new variables available: https://rez.readthedocs.io/en/stable/package_commands.html#pre-test-commands.

maxnbk commented:
@Ni-g-3l Just re-pinging to see if Jean-Christophe's command can help you with the CLA issue. I'd love to get this merged.

Ni-g-3l committed:
Add base and root bindings to pre test command

Signed-off-by: Nig3l <cots.max@gmail.com>

Ni-g-3l committed:
Add test coverage for root and base bindings in pre_test_commands

Add a pre_test_commands function to the testing_obj fixture package
that uses the root and base bindings, and a test entry that verifies
the bindings are available. Update affected test counts in test_test.py.

Signed-off-by: Nig3l <cots.max@gmail.com>

Ni-g-3l committed:
Update pre test documentation

Signed-off-by: Nig3l <cots.max@gmail.com>

Ni-g-3l commented:
Should be okay now :) sorry for late responding I was busy last week :)
